### DISCUSSION TITLE
Is it possible to translate PostgREST queries to SQL inside of Postgres?

### DISCUSSION THREAD

**Discussion by ChuckJonas** [PostgREST/postgrest#3817] [Q&A] [answered]
Does postgREST expose a SQL function that would allow you to translate a PostgREST query to a SQL query?

Basically, perform this part of the execution pipeline:   

_Reference from [Architecture docs_](https://docs.postgrest.org/en/v12/explanations/architecture.html) 
<img width="461" alt="Architecture_—_PostgREST_12_2_documentation" src="https://github.com/user-attachments/assets/ee4e0fe1-7168-44f8-b8a2-6cd396060d07" />

Or the opposite, the [supabase SQL -> Rest API translator](https://supabase.com/docs/guides/api/sql-to-rest) (but executed from SQL).

```sql
SELECT * FROM translatePostgREST(
    'books?select=title,description&description=ilike.*cheese*&order=title.desc&limit=5&offset=10'
)

-- would produce:
select
  title,
  description
from
  books
where
  description ilike '%cheese%'
order by
  title desc
limit
  5
offset
  10
```

**Comment by wolfgangwalther** [comment, marked as answer]
> Does postgREST expose a SQL function that would allow you to translate a PostgREST query to a SQL query?

No, this translation is done via Haskell, not SQL, so we don't have a function to expose.

You could enable statement logging to see which queries PostgREST is running: https://docs.postgrest.org/en/v12/references/observability.html#database-logs

If you just want to see execution plans, you can do it like this: https://docs.postgrest.org/en/v12/references/observability.html#execution-plan
